File: app/src/main/java/com/urt1rs/donkeywiki/entity/employee/Employee.java
package com.urt1rs.donkeywiki.entity.employee;

import com.urt1rs.donkeywiki.constant.employee.Position;

import java.util.List;

/**
 * @author Hongzhi.Liu
 * @date 2019/8/23
 */
public class Employee {

    protected String name;

    /**
     * 攻击
     */
    protected int attack;
    protected int defend;
    protected int block;
    /**
     * 星级
     */
    protected int star;
    /**
     * 招募标签
     */
    protected List<String> tags;
    /**
     * 定位
     */
    protected Position position;

    public Employee() {

    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Employee)) {
            return false;
        }
        return name.equals(((Employee) o).name);
    }

    @Override
    public int hashCode() {
        return name.hashCode();
    }



}

File: app/src/main/java/com/urt1rs/donkeywiki/entity/employee/Employee.kt
package com.urt1rs.donkeywiki.entity.employee

import com.urt1rs.donkeywiki.constant.employee.Position

/**
 * @author Hongzhi.Liu
 * @date 2019/8/23
 */
open class Employee {

    protected var name: String? = null

    /**
     * 攻击
     */
    protected var attack: Int = 0
    protected var defend: Int = 0
    protected var block: Int = 0
    /**
     * 星级
     */
    protected var star: Int = 0
    /**
     * 招募标签
     */
    protected var tags: MutableList<String>? = null
    /**
     * 定位
     */
    protected var position: Position? = null

    override fun equals(o: Any?): Boolean {
        return if (o !is Employee) {
            false
        } else name == o.name
    }

    override fun hashCode(): Int {
        return name!!.hashCode()
    }


}
